### DISCUSSION TITLE
[BUG]Get-Command -Module PnP.PowerShell doesn't return anything

### DISCUSSION THREAD

**Discussion by munrobasher** [pnp/powershell#4595] [General]
### Reporting an Issue or Missing Feature
Issue

### Expected behavior 
List the cmdlets in the module

### Actual behavior 
Nothing is listed

### Steps to reproduce behavior
Open PowerShell v5.1 and run **Get-Command -Module PnP.PowerShell**

### What is the version of the Cmdlet module you are running?
2.12.0

### Which operating system/environment are you running PnP PowerShell on?
- [X] Windows

### Supplemental information. 
If you run Get-Command -Module PnP.PowerShell _after_ executing a PnP.PowerShell command (e.g. Connect-PnPOnline), then the cmdlets are listed.

![Q3ktGPxLxe](https://github.com/user-attachments/assets/95d7ef15-4371-4d11-89cb-ac015b4304d2)

**Comment by jackpoz** [comment]
Hi, a few things to notice:
- https://github.com/pnp/powershell?tab=readme-ov-file#important---new-pnp-powershell-2x mentions that PnP.PowerShell 2.x doesn't support Windows PowerShell 5.1 . Only PowerShell 7.x is supported, we latest 2.99.x nightly builds supporting PowerShell 7.4.4+ as mentioned at https://github.com/pnp/powershell/blob/dev/MIGRATE-2.0-to-3.0.md#updating-from-pnp-powershell-2x-to-3x
- Your screenshot shows "1.12.0", even if the version you wrote is 2.12.0
- `Get-Command` is an out of the box PowerShell command and there is little that PnP PowerShell team can do to fix issues in it
- Works fine on PowerShell 7.4.6 and PnP PowerShell nightly build 2.99.60
![image](https://github.com/user-attachments/assets/3ed061bd-7167-409c-bdc8-9c0b592c778c)

You could edit the `$profile` script to import the module automatically, that way you won't need to invoke any command to load it and Get-Command should work right away.

**Comment by munrobasher** [comment]
This indeed could be problems with versions... the above is running on an Windows 11 VM that's been around for a while and will have had the module installed before. When I ran it yesterday, I got the warning about new version so I updated it. I suspect that VM is now very confused... you're right that Get-Command when it works shows 1.12 but **Get-InstalledModule -Name Pnp.PowerShell** indicates 2.12.0 is installed (which is why I said 2.12.0):

![image](https://github.com/user-attachments/assets/5a34f0ed-e947-4d75-8c21-7b3b430349ba)

I've got hundreds of PowerShell scripts and not ready to switch to PS 7 just yet. The fact that -ClientId is working on this older version, I might stick with latest v1 for now until I've got time to re-visit.

This also explains why it's not working on my new main PC - installed the module but Connect-PnpOnline isn't available at all - Import-Module gave a better error message about incompatible versions. I rolled my PC back as I don't like quirks like this.

Is the warning message about updates baked into the code or is it coming from the internet? If the later, then a note about PowerShell versions wouldn't go amiss. 

The VM is backed up - just going to restore a backup and repeat.
